### PR TITLE
Add GMS implementation

### PR DESCRIPTION
**Pull request by FssAy**
Sessionless for Game Maker Studio 2.3 as an extension.

I remember there was some talk to add Unity extension for Sessionless, so I though it would be nice to also create one for GMS.

Currently it only works on Windows, but more platforms are possible to support. Due to the nature of GMS it was necessary to create an external library that does all the algorithmic operations under the hood. _(Creating the secp256k1 implementation raw in GML would be extremely painful and inefficient)_

Docs can be found at:
- src/gms/README.md
- src/gms/project/README.md

I will publish the package to the marketplace after the merge.

### CODE REVIEW

File: src/gms/lib-rust/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#[macro_use]
+extern crate lazy_static;
+
+use async_load::types::*;
+use sessionless::{PrivateKey, Sessionless};
+use sessionless::hex::{FromHex, IntoHex};
+
+#[no_mangle]
+#[allow(unused_variables)]
+pub unsafe extern "C" fn slap_ctx_new() -> GMLDouble {
```
Comment (zach-planet-nine) [inline src/gms/lib-rust/src/lib.rs:12]: 😄 love the name of this fn

File: src/gms/project/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+2. Create `Sessionless` context object.
+3. Remember to call `Clean()` method whenever the context object is destroyed.
+
+## `Sessionless`
+Sessionless context object is an object that lets you communicate with the underlying engine.
+In most cases you don't have to create it more than once in the entire lifetime of the process.
+It's allocated globally on the heap, and it is your responsibility to deallocate it whenever it's no longer used.
+
+The context object can be created as follows:
+```js
```
Comment (zach-planet-nine) [inline src/gms/project/README.md:14]: is this the right syntax type?

**Review by zach-planet-nine — APPROVED** [review]
love this. Thank you for doing this. I think to keep naming consistent with the repo, I would rename the src/gms folder to src/gml. Small change, and kind of pedantic since gms is the only thing that uses gml, but something could adopt it in the future, and this might still work.

File: src/gms/project/README.md (hunk 2 above)
Comment (FssAy) [reply src/gms/project/README.md:14]: There is no support for GML, but the JS syntax have the best coloring scheme

**Pull request merged**